stacktest/fakemetrics/out/kafkamdm/kafkamdm.go

```diff
@@ -46,11 +47,11 @@ func (p *LastNumPartitioner) Partition(m schema.PartitionedMetric, numPartitions
 }
 
 // key is by metric name, but won't be used for partition setting
-func (p *LastNumPartitioner) GetPartitionKey(m schema.PartitionedMetric, b []byte) ([]byte, error) {
-	return m.KeyBySeries(b), nil
+func (p *LastNumPartitioner) GetPartition(m schema.PartitionedMetric, numPartitions int32) (int32, error) {
+	return m.PartitionID(schema.PartitionBySeries, numPartitions)
 }
 
-func New(topic string, brokers []string, codec string, stats met.Backend, partitionScheme string) (*KafkaMdm, error) {
+func New(topic string, brokers []string, codec string, stats met.Backend, partitionScheme string, numPartitions int32) (*KafkaMdm, error) {
```

Comment:
Why do we need numPartitions. Shouldn't we just be querying kafka for the number of partitions?

Reply:
Yes